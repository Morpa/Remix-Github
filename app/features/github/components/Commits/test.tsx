import { render, screen } from '@testing-library/react'
import { Commits } from '.'

const props = {
  user: {
    login: 'login',
    avatar_url: 'https://avatars3.githubusercontent.com/u/1234?v=4',
    html_url: 'www.google.com',
    bio: 'bio'
  },
  commits: [
    {
      sha: 'sha',
      message: 'hello',
      html_url: 'www.github.com'
    }
  ]
}

describe('Commits', () => {
  it('should render correctly ', () => {
    const { container } = render(<Commits {...props} />)

    expect(
      screen.getByRole('heading', { name: 'Activity' })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('link', {
        name: /view all activity/i
      })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render with the passed props', () => {
    render(<Commits {...props} />)

    expect(
      screen.getByRole('link', {
        name: /login/i
      })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('img', {
        name: /login/i
      })
    ).toBeInTheDocument()

    expect(screen.getByText(/hello/i)).toBeInTheDocument()

    expect(
      screen.getByRole('heading', {
        name: /login/i
      })
    ).toBeInTheDocument()
  })
})
