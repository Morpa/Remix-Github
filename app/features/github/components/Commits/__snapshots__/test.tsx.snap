// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Commits should render correctly  1`] = `
<div
  class="pl-6 lg:w-80"
>
  <div
    class="pt-6 pb-2"
  >
    <h2
      class="text-sm font-semibold"
    >
      Activity
    </h2>
  </div>
  <div>
    <ul
      class="divide-y divide-gray-200"
      role="list"
    >
      <li
        class="py-4"
      >
        <a
          href="www.github.com"
          rel="noreferrer"
          target="_blank"
        >
          <div
            class="flex space-x-3"
          >
            <img
              alt="login"
              class="h-6 w-6 rounded-full"
              src="https://avatars3.githubusercontent.com/u/1234?v=4"
            />
            <div
              class="flex-1 space-y-1"
            >
              <h3
                class="text-sm font-medium"
              >
                login
              </h3>
              <p
                class="text-sm text-gray-500"
              >
                hello
              </p>
            </div>
          </div>
        </a>
      </li>
    </ul>
    <div
      class="py-4 text-sm border-t border-gray-200"
    >
      <a
        class="text-indigo-600 font-semibold hover:text-indigo-900"
        href="#"
      >
        View all activity 
        <span
          aria-hidden="true"
        >
          â†’
        </span>
      </a>
    </div>
  </div>
</div>
`;
